benchmarks/htap/lib/stats.py

```diff
@@ -68,18 +96,34 @@ def __init__(self, dsn, num_oltp_slots, num_olap_slots, csv_interval, ignored_qu
         self.csv_oltp = None
         self.csv_dbstats = None
         self.conn = None
+        self.history_length = history_length
+
+    def get_history_length(self):
+        return len(self.data['oltp']['history'])
 
     def _update_oltp_stats(self, stats):
         for stat in stats:
-            self.data['oltp'][stat['query']].append(stat)
             self.data['oltp_counts'][stat['query']][stat['status']] += 1
 
-
-    def cleanup_oltp_stats(self, time_now):
-        cutoff = int(time_now - HISTORY_LENGTH)
-        for query_type in self.data['oltp'].keys():
-            # keep HISTORY_LENGTH of history
-            self.data['oltp'][query_type] = [s for s in self.data['oltp'][query_type] if s['timestamp'] >= cutoff]
+            oltp = self.data['oltp']
+            second = int(stat['timestamp'])
+            if second < oltp['history'][0][0]:
+                # Too old sample, discard it
```

Comment:
this should not be possible, or? add a warning maybe?

Reply:
I haven't seen it, but if a single OLTP worker falls a lot behind for any reason it might happen, better cover the case than failing or wrong stats. I'll add a warning